### PR TITLE
Rd 40 create all classes

### PR DESCRIPTION
**Pull request by NazarShchur**
created all classes with hibernate relationships and lombok annotations

### CODE REVIEW

**Comment by SashaRoyenko** [comment]
Could you make new lines after fields, please?

File: src/main/java/com/robosh/ejournal/entity/HomeTask.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import java.time.LocalDate;
+
+@Data
+@NoArgsConstructor
+@Builder
+@Entity
```
Comment (Ostape) [inline src/main/java/com/robosh/ejournal/entity/HomeTask.java:16]: I think we need to put  argument in @Entity(name="home_task") in order if we will want to change class name. Need to be discussed with all team members.

File: src/main/java/com/robosh/ejournal/entity/Schedule.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
```
Comment (Ostape) [inline src/main/java/com/robosh/ejournal/entity/Schedule.java:15]: Maybe we need to add @AllArgsConstructor for all classes? Need to be discussed with all team members

File: src/main/java/com/robosh/ejournal/entity/Teacher.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.robosh.ejournal.entity;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import javax.persistence.Entity;
+
```
Comment (Ostape) [inline src/main/java/com/robosh/ejournal/entity/Teacher.java]: remove this empty string. Always use ctrl+alt+L to apply code style and ctrl+alt+O to remove redundant  imports

File: src/main/java/com/robosh/ejournal/entity/Mark.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package com.robosh.ejournal.entity;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
```
Comment (Ostape) [inline src/main/java/com/robosh/ejournal/entity/Mark.java]: remove this import with wildcard. Need to be discussed with team about code style: with what number imports we need to use wildcard imports.

File: src/main/java/com/robosh/ejournal/entity/School.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.robosh.ejournal.entity;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
```
Comment (Ostape) [inline src/main/java/com/robosh/ejournal/entity/School.java]: why we have here two empty lines? In the next file we have only one empty string between different package imports. Please use ctrl+alt+L in all files

File: src/main/java/com/robosh/ejournal/entity/User.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    protected String lastName;
+
+    protected String email;
+
+    protected String password;
+
+    protected String telephone;
+
+    @ManyToOne
+    private School school;
```
Comment (Ostape) [inline src/main/java/com/robosh/ejournal/entity/User.java]: why this field is private? maybe we need to make it protected?

**Review by Ostape — DISMISSED** [review]

File: src/main/java/com/robosh/ejournal/entity/Schedule.java (hunk 2 above)
Comment (NazarShchur) [reply src/main/java/com/robosh/ejournal/entity/Schedule.java:15]: All args constructor is included in @Data

File: src/main/java/com/robosh/ejournal/entity/User.java (hunk 6 above)
Comment (NazarShchur) [reply src/main/java/com/robosh/ejournal/entity/User.java]: Probably

**Review by SashaRoyenko — CHANGES_REQUESTED** [review]
Make approves and I will merge

File: src/main/java/com/robosh/ejournal/entity/Schedule.java (hunk 2 above)
Comment (SashaRoyenko) [reply src/main/java/com/robosh/ejournal/entity/Schedule.java:15]: > All args constructor is included in @DaTa
It contains @@RequiredArgsConstructor. Here is a link to difference https://projectlombok.org/features/constructor. But for now, I think it's ok. We will add it if we needed

**Review by SashaRoyenko — APPROVED** [review]

**Review by Ostape — APPROVED** [review]

**Pull request merged**